common/util/FileUtil.cpp

```diff
@@ -0,0 +1,36 @@
+#include "FileUtil.h"
+#include <iostream>
+#include <stdio.h>  /* defines FILENAME_MAX */
+
+#ifdef _WIN32
+  #include <direct.h>
+  #define GetCurrentDir _getcwd
+#else
+  #include <unistd.h>
+  #define GetCurrentDir getcwd
+#endif
+
+std::string FileUtil::GetCurrentWorkingDir() {
+  char buff[FILENAME_MAX];
+  GetCurrentDir(buff, FILENAME_MAX);
+  std::string current_working_dir(buff);
+  return current_working_dir;
+}
+
+std::string FileUtil::get_file_path(std::vector<std::string> input) {
+  std::string currentPath = FileUtil::GetCurrentWorkingDir();  // std::filesystem::current_path();
+  char dirSeparator;
+
+  #ifdef _WIN32
+    dirSeparator = '\\';
+  #else
+    dirSeparator = '/';
+  #endif
+
+  std::string filePath = currentPath;
+    for (int i = 0; i < input.size(); i++) {

```

Comment:
It looks like the formatting here is slightly off - we use a tool called `clang-format` to automatically format the code and there's a test that checks for this.  There's probably a way to run `clang-format` to format your code within visual studio.

Reply:
Thanks, I've got `clang-format` working and have run it on all files. Very helpful because VS kept reformatting things strangely on its own!